components/PopupList.js

```diff
@@ -0,0 +1,63 @@
+import React, { useState } from 'react';
+import Modal from './Modal'
+import Form from './Dashboard/Assets/CreateForm'
+import Input from './Input'
+
+
+function PopupList({ isOpen, setIsOpen, list, renderRows, columns, title, create }) {
+    const [searchQuery, setSearchQuery] = useState('');
+
+    const handleSearchQueryChange = (event) => {
+        setSearchQuery(event.target.value);
+    };
+
+    const filteredList = searchQuery
+        ? list.filter((item) => {
+            if (typeof item === 'string') {
+                return item.toLowerCase().includes(searchQuery.toLowerCase());
+            } else {
+                if (item.name !== undefined) {
+                    if (typeof item.name === 'string')
+                        return item.name.toLowerCase().includes(searchQuery.toLowerCase());
+                }
+                else
+                    return item.id.toLowerCase().includes(searchQuery.toLowerCase());
```

Comment:
```suggestion
                if (typeof item.name === 'string') {
                    return item.name.toLowerCase().includes(searchQuery.toLowerCase());
                }
                else {
                    return item.id.toLowerCase().includes(searchQuery.toLowerCase());
                }
```

nao precisas de confirmar se é undefined e se é do tipo string pq ela nao pode ser undefined ou null e ser uma string ao mesmo tempo